tests/unit/test_help.py

```diff
@@ -1458,3 +1461,31 @@ def test_helprequested_command_format_truncated(parameters):
     """
     )
     assert str(raised.value) == expected
+
+
+class AppConfigCommand(BaseCommand):
+
+    name: str = "app-config"
+    help_msg: str = "Help text"
+    overview: str = "Overview"
+
+    def fill_parser(self, parser: ArgumentParser) -> None:
+        assert isinstance(self.config, dict)
+
+        choices = self.config["choices"]
+        parser.add_argument(
+            "number",
+            choices=choices,
+            help=f"The number to use. Possible values are {choices}.",
+        )
+
+
+def test_helprequested_command_app_config():
```

Comment:
suggestion: I understand this is only testing the `if global_args["help"]` branch in `pre_parse_args()`. I suggest adding one more test for the `if global_args["help"]` branch.

 Maybe convert this test to a parametrized one and provide `--help` and `help` as argument?

Reply:
good catch! Updated following you suggestion, thanks